### DISCUSSION TITLE
Interoperability with yarn v2's link?

### DISCUSSION THREAD

**Discussion by muryoh** [webpack/webpack#13456] [Q&A]
Hi guys,

I struggle to find the information I want - sorry if it's a silly question :-)

Basically, I want to be able to use [yarn link](https://yarnpkg.com/cli/link) with a project I have webpack setup for
However I am using yarn2, and unlike yarn 1 which was doing a symlink in the node_modules directly, this one basically just adds the following to my package.json:
```
  "resolutions": {
    "@mylinked-package": "portal:/home/muryoh/@mylinked-package"
  },
```

I cannot find how to have webpack properly consider that configuration
Is that supported in any way?

Cheers and thanks for the help :pray:

**Comment by alexander-akait** [comment]
webpack uses yarn API to get path to package https://github.com/webpack/enhanced-resolve/blob/main/lib/PnpPlugin.js

**Reply by muryoh** [reply]
Are you saying it should work out of the box?
I forgot to mention I'm using webpack v4 for now :see_no_evil:

**Reply by alexander-akait** [reply]
Time to move on webpack v5 :smile: Yep, out of box without any tricks, but we don't ahve strong portal tests, but I think should work due we use yarn API

**Reply by muryoh** [reply]
I see, but iiuc this uses the pnpapi and I think it's not initialized if not used in a "pnp environment", right?

**Reply by alexander-akait** [reply]
Do you use yarn 2 without pnp?

**Reply by alexander-akait** [reply]
Anyway you can create small example how you use it and I will look at this

**Reply by muryoh** [reply]
I just looked into it again and it does work :scream_cat: 

My problem was more yarn related than webpack related
When the package wasn't previously installed, doing a **yarn link** of said package doesn't do any linking in node_modules, doesn't modify the yarn.lock file, but only modifies the package.json file with the resolutions.. So obviously webpack couldn't know what to do then
